src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -459,10 +459,14 @@ bool layout_optimizer::can_fuse_reorder_to_prev(program_node& prev, program_node
             return true;
         };
 
+        // permute [fsv -> fsv] + reorder [fsv -> bfyx]
+        // prevent reorder for following cases for performance
+        // (1) moving f axis to last dim, because f is already continuous
+        // (2) permute is not rotating case, because rotating cases are good to fuse reorder since it can be mapped to tiled kernel
         auto& permute_order = prev.as<permute>().get_primitive()->permute_order;
         if ((fmt_prev == format::b_fs_yx_fsv4 || fmt_prev == format::b_fs_yx_fsv32 || fmt_prev == format::b_fs_zyx_fsv32 ||
          fmt_prev == format::b_fs_yx_fsv16 || fmt_prev == format::b_fs_zyx_fsv16 || fmt_prev == format::bs_fs_yx_bsv16_fsv16)
-         && permute_order[1] == 2
+         && (permute_order.back() == 1)
```

Comment:
Hmm, shouldn't it be `permute_order[2] == 1` for current axis order meaning that `f` becomes new `x` dimension after permutation? I think your condition (and most likely `==` should be changed to `!=`) should be applied after permute order alignment with IE

Reply:
Ah yeah, this is IE alignment. yes.. in the current version it shoud be permute_order[1] = last_dim.

Reply:
> permute_order[1] = last_dim

Why? It means that for bfyx layout `y` goes to `f` position, and this check will allow both `byxf` and `byfx` cases (and actually more cases with `b` dim transposition like xybf). but the intention is `// (1) moving f axis to last dim, because f is already continuous` which means `permute_order[2] == 1`.

Reply:
https://github.com/openvinotoolkit/openvino/blob/5fe10299ab711798f046cae530164bf2ee32132f/src/plugins/intel_gpu/include/intel_gpu/plugin/common_utils.hpp#L167 Here the IE order {0, 2, 3, 1} (which is rotating) is to be converted to cldnn order as {0, 3, 1, 2}.

Reply:
That's right, but  {0, 2, 1, 3} axis order (which is not rotating) is to be converted to {0, 3, 2, 1} and it'll also pass your check.

Reply:
0, 3, 2, 1 is the target of this condition to prohibit fusing. 
0, 3, 1, 2 (rotating) is okay to be fused with reorder because the fsv blocked format can support bfyx output.